  INFO root [2018-11-29 17:41:06,176] - 
  INFO root [2018-11-29 17:41:06,176] - -----------------------------------------------------------------------------------------


  INFO root [2018-11-29 17:41:06,176] - Resmgr::initialize: KBE_ROOT=E:/kbengine/
  INFO root [2018-11-29 17:41:06,177] - Resmgr::initialize: KBE_RES_PATH=E:/kbengine/kbe/res/;E:/kbengine/kbengine_frame_assets/;E:/kbengine/kbengine_frame_assets/scripts/;E:/kbengine/kbengine_frame_assets/res/
  INFO root [2018-11-29 17:41:06,177] - Resmgr::initialize: KBE_BIN_PATH=E:/kbengine/kbe/bin/server/
  INFO root [2018-11-29 17:41:06,181] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2018-11-29 17:41:06,181] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2018-11-29 17:41:06,184] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20875 )
  INFO root [2018-11-29 17:41:06,184] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:20875, SOMAXCONN=128.
  INFO root [2018-11-29 17:41:06,184] - server-configs:
  INFO root [2018-11-29 17:41:06,184] - 	internalTcpAddr : 192.168.56.1:20875
  INFO root [2018-11-29 17:41:06,184] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2018-11-29 17:41:06,184] - 	componentID : 9000
  INFO root [2018-11-29 17:41:07,841] - ---- loginapp Version: 2.3.0. ScriptVersion: 0.1.0. Protocol: AC120D8722168F72D5786E642068E11C. Config: Debug X86. Built: 11:06:44 Nov 12 2018. AppID: 9000. UID: 10103. PID: 8828 ----
  INFO root [2018-11-29 17:41:07,842] - Server : Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz with 2097151kB RAM
  INFO root [2018-11-29 17:41:07,842] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2018-11-29 17:41:07,842] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2018-11-29 17:41:07,977] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [2018-11-29 17:41:07,977] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [2018-11-29 17:41:07,978] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [2018-11-29 17:41:07,981] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [2018-11-29 17:41:07,981] - Script::install_py_dlls(): PyInit_select
 DEBUG root [2018-11-29 17:41:07,981] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [2018-11-29 17:41:07,982] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [2018-11-29 17:41:08,012] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [2018-11-29 17:41:08,062] - Script::install(): is successfully, Python=(3.4.2 (default, Nov 12 2018, 11:06:36) [MSC v.1915 32 bit (Intel)])!
  INFO root [2018-11-29 17:41:08,062] - Script::install(): KBExtra is successfully!
S_INFO root [2018-11-29 17:41:08,064] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2018-11-29 17:41:08,064] - TelnetServer server is running on port 31000
  INFO root [2018-11-29 17:41:08,427] - ---- loginapp is running ----
 DEBUG root [2018-11-29 17:41:08,429] - Components::process(): Request for the process of identity...
  INFO root [2018-11-29 17:41:15,564] - Components::findComponents: find logger(1)...
  INFO root [2018-11-29 17:41:15,652] - Components::findComponents: found logger, addr:192.168.56.1:20870
  INFO root [2018-11-29 17:41:15,652] - Components::addComponent: reset orderLog, uid=10103!
  INFO root [2018-11-29 17:41:15,652] - Components::addComponent[logger], uid=10103, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2018-11-29 17:41:15,653] - Components::findComponents: find dbmgr(2)...
  INFO root [2018-11-29 17:41:15,658] - Components::findComponents: found logger, addr:192.168.56.1:20870
  INFO root [2018-11-29 17:41:15,661] - Components::findComponents: found dbmgr, addr:192.168.56.1:20871
  INFO root [2018-11-29 17:41:15,661] - Components::addComponent: reset orderLog, uid=10103!
  INFO root [2018-11-29 17:41:15,661] - Components::addComponent[dbmgr], uid=10103, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2018-11-29 17:41:15,762] - Components::findComponents: found dbmgr, addr:192.168.56.1:20871
  WARN root [2018-11-29 17:41:16,763] - Forwarding logs to logger[192.168.56.1:20870/0/0/0]...
 ERROR root [2018-11-29 17:48:13,668] - Components::removeComponentByChannel: logger : 2000, Abnormal exit! TCPPacketReceiver::processRecv(): error=10054

  INFO root [2018-11-29 17:48:13,671] - The following logs sent to logger failed:
  WARN root [2018-11-29 17:48:13,671] - ==> [2018-11-29 17:48:13 668] TCPPacketReceiver::processPendingEvents(192.168.56.1:20870): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2018-11-29 17:48:13,671] - ==> [2018-11-29 17:48:13 668] Components::removeComponentByChannel: logger : 2000, Abnormal exit! TCPPacketReceiver::processRecv(): error=10054

  WARN root [2018-11-29 17:48:13,722] - TCPPacketReceiver::processPendingEvents(192.168.56.1:20871): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2018-11-29 17:48:13,723] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit! TCPPacketReceiver::processRecv(): error=10054

  INFO root [2018-11-29 17:48:13,727] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2018-11-29 17:48:13,733] - onLoginAppShutDown()
  INFO root [2018-11-29 17:48:13,733] - Shutdowner::onShutdown: shutting down
  INFO root [2018-11-29 17:48:13,733] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [2018-11-29 17:48:13,734] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [2018-11-29 17:48:13,734] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [2018-11-29 17:48:13,734] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [2018-11-29 17:48:13,734] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [2018-11-29 17:48:13,734] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [2018-11-29 17:48:13,734] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [2018-11-29 17:48:13,734] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [2018-11-29 17:48:13,735] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [2018-11-29 17:48:13,745] - Script::uninstall(): is successfully!
 DEBUG root [2018-11-29 17:48:13,745] - ThreadPool::destroy(): starting size 2.
  WARN root [2018-11-29 17:48:14,047] - ThreadPool::destroy(): waiting for thread(2)[0x22a9ec8,0x22a94a0,], try=1
  INFO root [2018-11-29 17:48:15,421] - 
  INFO root [2018-11-29 17:48:15,421] - -----------------------------------------------------------------------------------------


  INFO root [2018-11-29 17:48:15,422] - Resmgr::initialize: KBE_ROOT=E:/kbengine/
  INFO root [2018-11-29 17:48:15,422] - Resmgr::initialize: KBE_RES_PATH=E:/kbengine/kbe/res/;E:/kbengine/kbengine_frame_assets/;E:/kbengine/kbengine_frame_assets/scripts/;E:/kbengine/kbengine_frame_assets/res/
  INFO root [2018-11-29 17:48:15,422] - Resmgr::initialize: KBE_BIN_PATH=E:/kbengine/kbe/bin/server/
  INFO root [2018-11-29 17:48:15,429] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2018-11-29 17:48:15,429] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2018-11-29 17:48:15,431] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:21103 )
  INFO root [2018-11-29 17:48:15,432] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:21103, SOMAXCONN=128.
  INFO root [2018-11-29 17:48:15,432] - server-configs:
  INFO root [2018-11-29 17:48:15,432] - 	internalTcpAddr : 192.168.56.1:21103
  INFO root [2018-11-29 17:48:15,432] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2018-11-29 17:48:15,432] - 	componentID : 9000
  INFO root [2018-11-29 17:48:15,806] - Components::findLogger: found logger, addr:192.168.56.1:21098
  INFO root [2018-11-29 17:48:15,806] - Components::addComponent: reset orderLog, uid=10103!
  INFO root [2018-11-29 17:48:15,806] - Components::addComponent[logger], uid=10103, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2018-11-29 17:48:15,810] - ---- loginapp Version: 2.3.0. ScriptVersion: 0.1.0. Protocol: AC120D8722168F72D5786E642068E11C. Config: Debug X86. Built: 11:06:44 Nov 12 2018. AppID: 9000. UID: 10103. PID: 17116 ----
  INFO root [2018-11-29 17:48:15,810] - Server : Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz with 2097151kB RAM
  INFO root [2018-11-29 17:48:15,810] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2018-11-29 17:48:15,811] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2018-11-29 17:48:15,936] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [2018-11-29 17:48:15,936] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [2018-11-29 17:48:15,936] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [2018-11-29 17:48:15,939] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [2018-11-29 17:48:15,939] - Script::install_py_dlls(): PyInit_select
 DEBUG root [2018-11-29 17:48:15,940] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [2018-11-29 17:48:15,940] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [2018-11-29 17:48:15,974] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [2018-11-29 17:48:16,006] - Script::install(): is successfully, Python=(3.4.2 (default, Nov 12 2018, 11:06:36) [MSC v.1915 32 bit (Intel)])!
  INFO root [2018-11-29 17:48:16,007] - Script::install(): KBExtra is successfully!
S_INFO root [2018-11-29 17:48:16,008] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2018-11-29 17:48:16,009] - TelnetServer server is running on port 31000
  INFO root [2018-11-29 17:48:16,318] - ---- loginapp is running ----
 DEBUG root [2018-11-29 17:48:16,318] - Components::process(): Request for the process of identity...
  WARN root [2018-11-29 17:48:22,420] - Forwarding logs to logger[192.168.56.1:21098/0/0/0]...
 ERROR root [2018-11-29 17:51:11,896] - Components::removeComponentByChannel: logger : 2000, Abnormal exit! TCPPacketReceiver::processRecv(): error=10054

  INFO root [2018-11-29 17:51:11,898] - The following logs sent to logger failed:
  WARN root [2018-11-29 17:51:11,898] - ==> [2018-11-29 17:51:11 897] TCPPacketReceiver::processPendingEvents(192.168.56.1:21098): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2018-11-29 17:51:11,898] - ==> [2018-11-29 17:51:11 897] Components::removeComponentByChannel: logger : 2000, Abnormal exit! TCPPacketReceiver::processRecv(): error=10054

  WARN root [2018-11-29 17:51:11,960] - TCPPacketReceiver::processPendingEvents(192.168.56.1:21100): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2018-11-29 17:51:11,962] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit! TCPPacketReceiver::processRecv(): error=10054

  INFO root [2018-11-29 17:51:11,963] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2018-11-29 17:51:11,964] - onLoginAppShutDown()
  INFO root [2018-11-29 17:51:11,964] - Shutdowner::onShutdown: shutting down
  INFO root [2018-11-29 17:51:11,964] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [2018-11-29 17:51:11,965] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [2018-11-29 17:51:11,975] - Script::uninstall(): is successfully!
 DEBUG root [2018-11-29 17:51:11,975] - ThreadPool::destroy(): starting size 2.
  WARN root [2018-11-29 17:51:12,277] - ThreadPool::destroy(): waiting for thread(2)[0x21cdd88,0x21cde58,], try=1
  INFO root [2018-11-29 17:51:13,524] - 
  INFO root [2018-11-29 17:51:13,524] - -----------------------------------------------------------------------------------------


  INFO root [2018-11-29 17:51:13,525] - Resmgr::initialize: KBE_ROOT=E:/kbengine/
  INFO root [2018-11-29 17:51:13,525] - Resmgr::initialize: KBE_RES_PATH=E:/kbengine/kbe/res/;E:/kbengine/kbengine_frame_assets/;E:/kbengine/kbengine_frame_assets/scripts/;E:/kbengine/kbengine_frame_assets/res/
  INFO root [2018-11-29 17:51:13,525] - Resmgr::initialize: KBE_BIN_PATH=E:/kbengine/kbe/bin/server/
  INFO root [2018-11-29 17:51:13,536] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2018-11-29 17:51:13,537] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2018-11-29 17:51:13,544] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:21180 )
  INFO root [2018-11-29 17:51:13,544] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:21180, SOMAXCONN=128.
  INFO root [2018-11-29 17:51:13,544] - server-configs:
  INFO root [2018-11-29 17:51:13,545] - 	internalTcpAddr : 192.168.56.1:21180
  INFO root [2018-11-29 17:51:13,545] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2018-11-29 17:51:13,545] - 	componentID : 9000
  INFO root [2018-11-29 17:51:15,218] - ---- loginapp Version: 2.3.0. ScriptVersion: 0.1.0. Protocol: AC120D8722168F72D5786E642068E11C. Config: Debug X86. Built: 11:06:44 Nov 12 2018. AppID: 9000. UID: 10103. PID: 7172 ----
  INFO root [2018-11-29 17:51:15,232] - Server : Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz with 2097151kB RAM
  INFO root [2018-11-29 17:51:15,232] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2018-11-29 17:51:15,233] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2018-11-29 17:51:15,412] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [2018-11-29 17:51:15,413] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [2018-11-29 17:51:15,413] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [2018-11-29 17:51:15,416] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [2018-11-29 17:51:15,417] - Script::install_py_dlls(): PyInit_select
 DEBUG root [2018-11-29 17:51:15,417] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [2018-11-29 17:51:15,417] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [2018-11-29 17:51:15,450] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [2018-11-29 17:51:15,489] - Script::install(): is successfully, Python=(3.4.2 (default, Nov 12 2018, 11:06:36) [MSC v.1915 32 bit (Intel)])!
  INFO root [2018-11-29 17:51:15,489] - Script::install(): KBExtra is successfully!
S_INFO root [2018-11-29 17:51:15,491] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2018-11-29 17:51:15,491] - TelnetServer server is running on port 31000
  INFO root [2018-11-29 17:51:15,944] - ---- loginapp is running ----
 DEBUG root [2018-11-29 17:51:15,959] - Components::process(): Request for the process of identity...
